bot/handlers.py

```diff
@@ -515,6 +449,24 @@ def debt_sum(
                                            reply_markup=buttons.get_user_buttons(users))
         return States.DEBTOR_NAME
 
+    def cancel_button_handler(
+        self,
+        update: Update,
+        context: CallbackContext,
+    ):
+        data = update.callback_query.data
+        if data == menu_items.CANCEL:
+            context.user_data.pop(States.EXPENSE, None)
+            context.user_data.pop(States.DEBT, None)
+            event_token = context.user_data[States.CURRENT_EVENT_TOKEN]
+            event = self._splitwise.get_event_info(event_token)
+            update.callback_query.edit_message_text(f'\n\n {event.name}\nВыберете пункт меню:')
+            update.callback_query.edit_message_reply_markup(reply_markup=buttons.get_event_commands_keyboard())
+            update.callback_query.answer('Закончили')
+            return States.EVENT_ACTIONS
+        return None
```

Comment:
Maybe `END` or smth instead of None?

Reply:
Conversation handler not change state when we return None